crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1194,6 +1202,9 @@ impl<'source> Lexer<'source> {
                         self.state = State::AfterNewline;
                         Tok::Newline
                     } else {
+                        if let Some(fstring) = self.fstrings.current_mut() {
+                            fstring.try_end_format_spec(self.nesting);
+                        }
                         Tok::NonLogicalNewline
```

Comment:
We need to end the lexing of format spec at every non logical newline because we're still inside the curly braces. This might be error prone if in the future there's other place where we might need to emit a `NonLogicalNewline` but I think it's not that likely.

Another solution is to have a new `State::AfterNonLogicalNewline` which then will be checked before we lex for `FStringMiddle`/`FStringEnd` token:

```rust
pub fn next_token(&mut self) -> LexResult {
    if let Some(fstring) = self.fstrings.current() {
		if self.state.is_after_non_logical_newline() && fstring.is_in_format_spec(self.nesting) {
			fstring.try_end_format_spec(self.nesting);
		} else if !fstring.is_in_expression(self.nesting) {
			// ...
		}
	}
	// ...
```

Reply:
This involves other changes as well so I'm leaning towards solution 1.